src/nvim/buffer_defs.h

```diff
@@ -1296,6 +1298,13 @@ struct window_S {
   linenr_T w_redraw_bot;            // when != 0: last line needing redraw
   bool w_redr_status;               // if true statusline/winbar must be redrawn
   bool w_redr_border;               // if true border must be redrawn
+  bool w_redr_numcol;               // if true 'numbercolumn' must be redrawn
+
+  // remember what is shown in the fold/sign column for the currently drawn line
+  char w_redr_fold_text[10];
+  char w_redr_sign_text[9][5];
+  int w_redr_fold_attr;
+  int w_redr_sign_attr[9];
```

Comment:
Would it be possible to pack all this into a struct and stack allocate it in `drawscreen.c` and pass the struct as an argument to win_line. Similar to what we do with `sattrs`.

Reply:
Yes, something like that would be better. Perhaps dependant on how/if we refactor `win_line()` to account for `'numbercolumn'`.

Reply:
Would we rather stack allocate it in `drawscreen.c` or in `win_line()`? The scope is per `win_line()` call. Would also have to pass this to `build_stl_str_hl()`, reasonable?